zookeeper/source/zk_nn_client.cc

```diff
@@ -0,0 +1,939 @@
+				for (auto data_node :data_nodes) {
+
+					LOG(INFO) << CLASS_NAME << "Reading data node " << data_node;
+					int error_code;
+
+					std::vector<std::string> split_address;
+					boost::split(split_address, data_node, boost::is_any_of(":"));
+					assert(split_address.size() == 2);
+
+					auto data = std::vector<std::uint8_t>();
+					if (zk->get(HEALTH_BACKSLASH + data_node + STATS, data, error_code)) {
+						LOG(ERROR) << CLASS_NAME << "Getting data node stats failed with " << error_code;
+					}
+
+					zkclient::DataNodePayload * payload = (zkclient::DataNodePayload *) (&data[0]);
+
+					DatanodeInfoProto* dn_info = located_block->add_locs();
+					DatanodeIDProto* id = dn_info->mutable_id();
+					id->set_ipaddr(split_address[0]);
+					id->set_hostname("localhost"); // TODO: Fill out with the proper value
+					id->set_datanodeuuid("1234");
+					id->set_xferport(payload->xferPort);
+					id->set_infoport(50020);
+					id->set_ipcport(payload->ipcPort);
+				}
+				// Then store this block's information
+			}
+			size += block_size;
+		}
+	}
+
+
+	// ---------------------------------------- HELPERS ----------------------------------------
+
+	std::string ZkNnClient::ZookeeperPath(const std::string &hadoopPath){
+		std::string zkpath = NAMESPACE_PATH;
+		if (hadoopPath.size() == 0) {
+			LOG(INFO) << "real real bad news";
+		}
+		if (hadoopPath.at(0) != '/'){
+			zkpath += "/";
+		}
+		zkpath += hadoopPath;
+		if (zkpath.at(zkpath.length() - 1) == '/'){
+			zkpath.at(zkpath.length() - 1) = '\0';
+		}
+		return zkpath;
+	}
+
+	void ZkNnClient::set_file_info(HdfsFileStatusProto* status, const std::string& path, FileZNode& znode_data) {
+		HdfsFileStatusProto_FileType filetype;
+		// get the filetype, since we do not want to serialize an enum
+		switch(znode_data.filetype) {
+			case(0):
+				filetype = HdfsFileStatusProto::IS_DIR;
+				break;
+			case(1):
+				filetype = HdfsFileStatusProto::IS_DIR;
+				break;
+			case(2):
+				filetype = HdfsFileStatusProto::IS_FILE;
+				break;
+			default:
+				break;
+
+		}
+
+		FsPermissionProto* permission = status->mutable_permission();
+		// Shorcut to set permission to 777.
+		permission->set_perm(~0);
+		// Set it to be a file with length 1, "foo" owner and group, 0
+		// modification/access time, "0" path inode.
+		status->set_filetype(filetype);
+		status->set_path(path);
+		status->set_length(znode_data.length);
+		status->set_blocksize(znode_data.blocksize);
+		std::string owner(znode_data.owner);
+		std::string group(znode_data.group);
+		status->set_owner(owner);
+		status->set_group(group);
+
+		status->set_modification_time(znode_data.modification_time);
+		status->set_access_time(znode_data.access_time);
+		LOG(INFO) << CLASS_NAME << "Successfully set the file info ";
+	}
+
+	/**
+	 * Split the str by the delimiter and return a vector of the split words
+	*/
+		std::vector<std::string> ZkNnClient::split(const std::string &str, char delim) {
```

Comment:
We should make a util file that does things like splitting strings, making uuids, etc.